src/main/java/com/celements/cleverreach/MailingConfig.java

```diff
@@ -124,8 +134,43 @@ private MailingConfig(Builder builder) {
     return contentHtml;
   }
 
+  public String getContentHtmlCleanXml() {
+    final Tidy tidy = new Tidy();
+    tidy.setInputEncoding(StandardCharsets.UTF_8.name());
+    tidy.setOutputEncoding(StandardCharsets.UTF_8.name());
+    tidy.setWraplen(Integer.MAX_VALUE);
+    tidy.setPrintBodyOnly(true);
+    tidy.setXmlOut(true);
+    tidy.setQuoteNbsp(true);
+    tidy.setSmartIndent(false);
+    ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+    String xml = "";
+    try {
+      LOGGER.trace("string byte conversion equals [{}]", getContentHtml().equals(new String(
+          getContentHtml().getBytes("UTF-8"))));
+      tidy.parseDOM(new ByteArrayInputStream(getContentHtml().getBytes(
+          StandardCharsets.UTF_8.name())), outputStream);
+      xml = outputStream.toString(StandardCharsets.UTF_8.name());
+      LOGGER.trace("tidy.parseDOM: in length [{}], out length [{}]", getContentHtml().length(),
+          xml.length());
+    } catch (UnsupportedEncodingException uee) {
+      LOGGER.warn("Encoding not available: {}", StandardCharsets.UTF_8.name(), uee);
+    }
+    return xml;
+  }
+
   public @Nullable String getContentHtmlCssInlined() throws CssInlineException {
-    return getCssInliner().inline(getContentHtml(), css);
+    String cleaned = getContentHtmlCleanXml();
+    LOGGER.debug("Original and cleaned HTML are identical [{}]", cleaned.equals(getContentHtml()));
```

Comment:
equals check is always executed, please use `LogUtils#defer(Supplier)` to defer the execution until the message is built.
```
    LOGGER.debug("...", defer(() -> cleaned.equals(getContentHtml())));
```